File: Kotlin/src/replace.kt
fun replace(string: String, subString: String, newString: String): String {
    if (subString == "") return string
    var stringElse = string
    var resultElse = ""
    while (true) {
        val index = stringElse.indexOf(subString)
        if (index == -1) {
            resultElse += stringElse
            return resultElse
        }
        val subStringElse = stringElse.substring(0,index)
        val indexSubString = index + subString.length
        stringElse = stringElse.substring(indexSubString,stringElse.length)
        resultElse += subStringElse + newString
    }
}

fun main() {
    val result = replace("Hello <username> and bye!", "<username>", "Marcus")
    println(result)
}

File: Tasks_edited/replace.js
// Replace substring with newstr

function replace(str, substr,newstr){
  if(substr.length == 0){
    return str
  }
  else{
    replaced = ''
    do{
      const index = str.indexOf(substr); 
        if(index === -1){
          return replaced + str
        }
        else{
          const start = str.substring(0, index);
          const end = str.substring(index + substr.length, str.length)
          replaced += start + newstr + end
          return replaced
        }
    }while (true);
  };
};

const result = replace('Hello <username> and bye!', '<username>', 'Marcus')
console.log(result)
